src/PaymentChannel.js

```diff
@@ -0,0 +1,114 @@
+import BigNumber from 'bignumber.js';
+
+import paymentChannelStateMessages from './payment_channel_state_service_pb';
+
+export default class PaymentChannel {
+  constructor(channelId, web3, account, paymentChannelStateServiceClient, mpeContract) {
+    this._channelId = channelId;
+    this._web3 = web3;
+    this._account = account;
+    this._mpeContract = mpeContract;
+    this._paymentChannelStateServiceClient = paymentChannelStateServiceClient;
+    this._state = {
+      nonce: new BigNumber(0),
+      lastSignedAmount: new BigNumber(0),
+    };
+  }
+
+  get channelId() {
+    return this._channelId;
+  }
+
+  get nonce() {
+    return this.state.nonce;
+  }
+
+  get lastSignedAmount() {
+    return this.state.lastSignedAmount;
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  updateState(gstate) {
```

Comment:
Seems not used anywhere

Reply:
This would not be used within the sdk but by developers building `PaymentChannelManagementStrategy` which does caching of payment channel state locally.

Reply:
I suppose we need to design it carefully. For me it too complex to implement it correctly if there are three sources of data:
- blockchain
- service daemon
- and some external source

You need to guarantee that there is no race conditions between them. And guarantee it for external source in general will be impossible. 

I would instead add an abstraction of `ChannelProvider` or `ChannelStateProvider` and implement it differently for cases when caching is used or not. The difference is that this implementation will be responsible for providing correct state. And external parties can get updates to this state using listener interface. 

It is very abstract explanation and I think we don't need to implement it in this PR. Lets say that in this PR we don't support external state caches and we will design and implement it later.

Reply:
Agree. Deleting it for now and we can look it after the first implementation.

Reply:
Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/1adf2e3c86a29e3437e5216bb44a22f035277728)